addons/mail/static/src/discuss/core/public_web/discuss_sidebar_categories.js

```diff
@@ -87,25 +110,108 @@ export class DiscussSidebarCategories extends Component {
         thread.leave();
     }
 
-    /**
-     * @param {MouseEvent} ev
-     * @param {import("models").Thread} thread
-     */
-    openThread(ev, thread) {
+    /** @param {MouseEvent} ev */
+    openThread(ev) {
         markEventHandled(ev, "sidebar.openThread");
-        thread.setAsDiscussThread();
+        this.thread.setAsDiscussThread();
     }
+}
+
+export class DiscussSidebarCategory extends Component {
+    static template = "mail.DiscussSidebarCategory";
+    static props = ["category"];
+    static components = { Dropdown };
 
-    /**
-     *
-     * @param {import("models").DiscussAppCategory} category
-     */
-    toggleCategory(category) {
+    setup() {
+        super.setup();
+        this.store = useState(useService("mail.store"));
+        this.discusscorePublicWebService = useState(useService("discuss.core.public.web"));
+        this.hover = useHover(["root", "floating*"], () => {
+            this.floating.isOpen = this.hover.isHover;
+        });
+        this.floating = useDropdownState();
+    }
+
+    /** @returns {import("models").DiscussAppCategory} */
+    get category() {
+        return this.props.category;
+    }
+
+    get actions() {
+        return [];
+    }
+
+    toggle() {
         if (this.store.channels.status === "fetching") {
             return;
         }
-        category.open = !category.open;
-        this.discusscorePublicWebService.broadcastCategoryState(category);
+        this.category.open = !this.category.open;
+        this.discusscorePublicWebService.broadcastCategoryState(this.category);
+    }
+}
+
+export class DiscussSidebarQuickSearchInput extends Component {
+    static template = "mail.DiscussSidebarQuickSearchInput";
+    static props = ["state", "autofocus?"];
+
+    setup() {
+        super.setup();
+        this.store = useState(useService("mail.store"));
+        if (this.props.autofocus) {
+            useAutofocus({ refName: "root" });
+        }
+    }
+
+    get state() {
+        return this.props.state;
+    }
+}
+
+/**
+ * @typedef {Object} Props
+ * @extends {Component<Props, Env>}
+ */
+export class DiscussSidebarCategories extends Component {
+    static template = "mail.DiscussSidebarCategories";
+    static props = {};
+    static components = {
+        DiscussSidebarCategory,
+        DiscussSidebarChannel,
+        DiscussSidebarQuickSearchInput,
+        Dropdown,
+    };
+
+    setup() {
+        super.setup();
+        this.store = useState(useService("mail.store"));
+        this.discusscorePublicWebService = useState(useService("discuss.core.public.web"));
```

Comment:
`discussCorePublicWebService` ?

Reply:
Not a big deal, was already there. Besides, I would like to kill that service so it feels stupid to rename the variable in the meantime.